handlers/photo-tactile-svg/tactile_svg.py

```diff
@@ -123,60 +126,33 @@ def handle():
         logging.debug("Sending response")
         return response
 
+    # Initialize svg if either object detection
+    # or semantic segmentation is present
+    svg = draw.Drawing(dimensions[0], dimensions[1])
+
     if "ca.mcgill.a11y.image.preprocessor.objectDetection" in preprocessors:
+        preprocessor_names.append('Object detection')
         o = preprocessors["ca.mcgill.a11y.image.preprocessor.objectDetection"]
         g = preprocessors["ca.mcgill.a11y.image.preprocessor.grouping"]
-        u = preprocessors["ca.mcgill.a11y.image.preprocessor.grouping"]
         objects = o["objects"]
         grouped = g["grouped"]
-        ungrouped = u["ungrouped"]
-        svg = draw.Drawing(dimensions[0], dimensions[1])
+        ungrouped = g["ungrouped"]
         layer = 0
-        if (len(grouped) > 0):
-            for i in range(len(grouped)):
-                ids = grouped[i]["IDs"]
-                category = objects[ids[0]]["type"]
-                layer += 1
-                g = draw.Group(data_image_layer="Layer " +
-                               str(layer), aria_label=category)
-                for j in range(len(ids)):
-                    # print(ids[j])
-                    x1 = int(objects[ids[j]]['dimensions'][0] * dimensions[0])
-                    x2 = int(objects[ids[j]]['dimensions'][2] * dimensions[0])
-                    y1 = int(objects[ids[j]]['dimensions'][1] * dimensions[1])
-                    y2 = int(objects[ids[j]]['dimensions'][3] * dimensions[1])
-                    width = abs(x2 - x1)
-                    height = abs(y2 - y1)
-                    start_y1 = abs(dimensions[1] - y1 - height)
-                    g.append(
-                        draw.Rectangle(
-                            x1,
-                            start_y1,
-                            width,
-                            height,
-                            stroke="#ff4477",
-                            stroke_width=2.5,
-                            fill="none",
-                            aria_label=category+" "+str(j+1)))
-
-                svg.append(g)
-
-        if (len(ungrouped) > 0):
-            for i in range(len(ungrouped)):
-                category = objects[ungrouped[i]]["type"]
-                layer += 1
-                x1 = int(objects[ungrouped[i]]
-                         ['dimensions'][0] * dimensions[0])
-                x2 = int(objects[ungrouped[i]]
-                         ['dimensions'][2] * dimensions[0])
-                y1 = int(objects[ungrouped[i]]
-                         ['dimensions'][1] * dimensions[1])
-                y2 = int(objects[ungrouped[i]]
-                         ['dimensions'][3] * dimensions[1])
+        for i in range(len(grouped)):
```

Comment:
FYI, a more _python_ way of doing this would be to do something like
```
for i, group in enumerate(grouped):
    ids = group["IDs"]
    # Other stuff
    aria_label = category + " " + str(i+1)
```